packages/manager/src/components/VLANSelect.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState } from 'react';
+
+import { useVLANsInfiniteQuery } from 'src/queries/vlans';
+
+import { Autocomplete } from './Autocomplete/Autocomplete';
+
+import type { Filter } from '@linode/api-v4';
+import type { SxProps, Theme } from '@mui/material';
+
+interface Props {
+  /**
+   * Disabled the Select
+   * @default false
+   */
+  disabled?: boolean;
+  /**
+   * An error that will show below the select
+   */
+  errorText?: string;
+  /**
+   * Default API filter
+   */
+  filter?: Filter;
+  /**
+   * Is called when a VLAN is selected
+   */
+  onChange?: (label: null | string) => void;
+  /**
+   * Optional Styles
+   */
+  sx?: SxProps<Theme>;
+  /**
+   * The label of the selected VLAN
+   */
+  value: null | string;
+}
+
+/**
+ * A VLAN select component that has the following features
+ * - Infinitly loads VLANs
```

Comment:
Such a small nit but since this component might be used elsewhere... 😅

```suggestion
 * - Infinitely loads VLANs
```

Reply:
Good catch, thank you!